### PR TITLE
Fix name/package parameter convert when passed list of values using with_items

### PR DESCRIPTION
**Pull request by dimid**
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
Packaging plugin for gentoo: packaging/os/portage.py

##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY
task file:

```
% cat roles/mysql/tasks/main.yml 
---
- name: check mysql packages
  package: name={{ item }} state=present
  with_items:
      - "{{ mysql_server_package }}"
      - "{{ mysql_python_package }}"
```

Ansible output:
```
failed: [servername] (item=[u'mysql', u'mysql-python']) => {
    "cmd": [
        "/usr/bin/emerge",
        "--ask=n",
        "['mysql'",
        " 'mysql-python']"
    ],
    "failed": true,
    "invocation": {
        "module_args": {
            "changed_use": false,
            "deep": false,
            "depclean": false,
            "getbinpkg": false,
            "name": [                                                                                                                                      
                "mysql", 
                "mysql-python"
            ], 
            "newuse": false, 
            "nodeps": false, 
            "noreplace": false, 
            "oneshot": false, 
            "onlydeps": false, 
            "package": "['mysql', 'mysql-python']", 
            "quiet": false, 
            "state": "present", 
            "sync": null, 
            "update": false, 
            "usepkg": false, 
            "usepkgonly": false, 
            "verbose": false
        }
    }, 
    "item": [
        "mysql", 
        "mysql-python"
    ], 
    "msg": "Packages not installed.", 
    "rc": 1, 
    "stderr": "!!! '['mysql'' is not a valid package atom.\n!!! Please check ebuild(5) for full details.\n", 
    "stdout": "", 
    "stdout_lines": []
}
```

List parameter is converted to string and then split by ',', so we get list with values "['mysql'" and "'mysql-python']". In fix I've just set explicit parameter type to 'list' and remove split by ',' because in case of explicit type it will be always a list.

Now output looks like:
```
ok: [servername] => (item=[u'mysql', u'mysql-python']) => {
    "changed": false,
    "invocation": {
        "module_args": {
            "changed_use": false,        
            "deep": false,
            "depclean": false,
            "getbinpkg": false,
            "name": [
                "mysql",
                "mysql-python"
            ],
            "newuse": false,
            "nodeps": false,
            "noreplace": false,
            "oneshot": false,
            "onlydeps": false,
            "package": [           
                "mysql",
                "mysql-python"      
            ],                       
            "quiet": false,
            "state": "present",
            "sync": null, 
            "update": false, 
            "usepkg": false, 
            "usepkgonly": false, 
            "verbose": false
        }
    }, 
    "item": [
        "mysql", 
        "mysql-python"
    ], 
    "msg": "Packages already present."
}
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @dimid for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by dimid** [comment]
ready_for_review i think, it is simple fix

**Comment by gregdek** [comment]
Thanks @dimid. To the current maintainers, @wltjr, @mischapeters, @ericchou1, @gundalow, @qalthos, @privateip please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by wltjr** [comment]
shipit Changes make sense. Long term seems items is better than iteritems, which is removed in Python 3 and items was modified to have an iterator. Also the type change to list to avoid call to split. Seems to be the main one, and another slipped in the iteritems change which was already commited.

**Comment by gregdek** [comment]
Thanks again to @dimid for this PR, and thanks @wltjr, @mischapeters, @ericchou1, @gundalow, @qalthos, @privateip for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: network/a10/a10_server_axapi3.py [hunk 1]
```diff
@@ -249,4 +249,3 @@ def main():
 
 if __name__ == '__main__':
     main()
-
```
Comment (gundalow) [inline network/a10/a10_server_axapi3.py]: Please revert this change. It's confusing to see it in this PR

**Comment by dimid** [comment]
@gundalow, i've cleaned up, only my changes is here right now.

**Comment by wltjr** [comment]
Sorry for missing that if I did when commenting to ship. Must have missed it in the modified files list in the original.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged, thanks.
